src/transformState.js

```diff
@@ -4,8 +4,39 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+const ADD_ACTION = 'addProperties';
+const REMOVE_ACTION = 'removeProperties';
+const CLEAR_ACTION = 'clear';
+
 function transformState(state, actions) {
   // write code here
+  let newState = state;
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ADD_ACTION:
+        newState = Object.assign(state, action.extraData);
+        break;
+
+      case REMOVE_ACTION:
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case CLEAR_ACTION:
+        for (const key in newState) {
+          delete [key];
```

Comment:
What are you deleting here?

Reply:
key of newState

Reply:
e.g. key === 'hello'

delete ['hello']

How does this relate to the state?